src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,134 @@
+                String searchTerm = queryParts[1];
+                mollyResponse.append(taskList.findTerm(searchTerm)).append("\n");
+                //return mollyResponse.toString();
+            } else {
+                throw new MollyException("Invalid find command. Please use 'find (search term)'.");
+            }
+
+        } else if (!userInput.toLowerCase().equals("list")) {
+            if (userInput.startsWith("todo")) {
+                if (userInput.equals("todo")) {
+                    throw new MollyException("Yikes! Sorry, the todo description cannot be empty.");
+                } else {
+                    Task newToDo = new Task(userInput.substring(5));
+                    mollyResponse.append(taskList.addTask(newToDo));
+                    storage.saveTasks(taskList);
+                }
+            } else if (userInput.startsWith("deadline")) {
+                if (userInput.equals("deadline")) {
+                    throw new MollyException("Yikes! Sorry, the description of a deadline cannot be empty.");
+                } else {
+                    String[] deadlineDetails = userInput.substring(9).split( " /by ");
+                    if (deadlineDetails.length == 2) {
+                        String description = deadlineDetails[0].trim();
+                        String by = deadlineDetails[1].trim();
+                        Deadline newDeadline = new Deadline(description, by);
+                        mollyResponse.append(taskList.addTask(newDeadline));
+                        storage.saveTasks(taskList);
+                    } else {
+                        Ui.printLine();
+                        throw new MollyException("Sorry, invalid format for deadline. Please follow the format: deadline (description) /by (end date). The end date can be in the format DD-MM-YYYY HHmm (24 hour format).");
+                    }
+                }
+            } else if (userInput.startsWith("event")) {
+                if (userInput.equals("event")) {
+                    throw new MollyException("Yikes! Sorry, the description of an event cannot be empty.");
+                } else {
+                    String[] eventDetails = userInput.substring(6).split( " /from | /to ");
+                    if (eventDetails.length == 3) {
+                        String description = eventDetails[0].trim();
+                        String from = eventDetails[1].trim();
+                        String to = eventDetails[2].trim();
+                        Event newEvent = new Event(description, from, to);
+                        mollyResponse.append(taskList.addTask(newEvent));
+                        storage.saveTasks(taskList);
+                    } else {
+                        throw new MollyException("Sorry, invalid format for event. Please follow the format: event (description) /from (start date or time) /to (end date or time). The start and end date/times can be in the format DD-MM-YYYY HHmm (24 hour format)");
```

Comment:
As others have mentioned, you can consider breaking these long lines up into shorter ones to keep within the 120 length limit.